webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/policies/Directive.scala

```diff
@@ -37,12 +37,77 @@
 
 package com.normation.rudder.domain.policies
 
+import com.normation.GitVersion.RevId
+
 import scala.xml._
 import com.normation.cfclerk.domain.TechniqueVersion
 import com.normation.cfclerk.domain.SectionSpec
 
+/*
+ * Two way of modeling the couple (directiveId, revId) :
+ * - either we do DirectiveId(uuid: String, revId: RevId).
+ *   We tell that the revision is part of the directive identifier, and that an identifier
+ *   always has a revision.
+ *   It is likely the solution that will make simpler to never forget the migration from
+ *   id = uuid to id = (uuid, rev). But:
+ *   - it may complexify some automatic derivation, for ex for JSON (ie either we change json format for
+ *     "directive": { "id": {"uuid":"xxxxx", "rev": "xxxxx" }, ... }
+ *     which is breaking change for no good reason, especially if we want to make "rev" (or "revId") optionnal
+ * - or we just add a "revId" (or "rev") field in directive. It doesn't break any existing serialisation API, it
+ *   allows to continue to speak about "the directive ID" as just the uuid part, and have rev in addition.
+ *   But it means that we will need to be extra-careful to not forget a place that for now use "id" and that will
+ *   need to be duplicated or augmented with an optionnal "rev" parameter.
+ *
+ * I *think* we should change as little serialisation format as possible, especially in API, because they are the worst
+ * breaking changes possible, and from an external observer, not providing rev should let everything work as before.
+ * So we should make a hard constraint that JSON API will remain the same (with possibly an added "rev" field).
+ *
+ * Given that, we should (at least in the begining) try to minize distance between API serialisation format and internal
+ * format. So go for the second option, and be careful when evolving methods.
+ *
+ * ==== some more evolution
+ *
+ * - forcing rev everywhere, when we want it to be almost always `defaultRev`, is not efficient. We should use
+ *   Option[RevId]
+ *   BUT it means that special attention need to be used on unserialisation: defaultValue (if serialised)
+ *   must be unserialized to `None`.
+ * - the code is crying for a DirectiveRId(id: DirectiveId, revId: Option[RevId] = None)
```

Comment:
With some more though, I think we should make that change. 
With that PR, the id is really the product of the two parts, and actually the `id` is the two, and we need to find a short, clear name for that tuple.
Also, I'm not thinking of the version par as a revision. Even after all the code and all, it's a "commit id" in my head. I'm not sure it's a good idea to try to use something else to talk about it.

Reply:
I think the directiveId should evolve to be a case class that includes the commit 

like 

case class DirectiveId { value : String, commit : Commit ) 

with Commit being a trait with two values Head and CommitId (value) , so that we don't use an Option, but something with a meaning

Reply:
I need to sum-up things here, but I agree that the ID is really two parts: an UUID and an optionnal commit id. 
I'm not sure it's a `head|commit`. The reallity for now is that it's optionnal (sometime, we have it, sometime no). But it could be dealt with at system boundaries (read from/to git, unserialize things, api, etc). And we would have a clearer internal model.

Reply:
We chose `revision`